### ISSUE TITLE
Lfs upload file cannot be ended

### ISSUE THREAD

**Issue by alvin6233** [go-gitea/gitea#28972]
### Description

I use lfs upload file about 1G. But it shows that it keeps uploading and the uploaded size exceeds the actual size. 
The phenomenon is as follows：
```
(base) ➜  didi_2 git:(main) ✗ git push
Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.https://code.org.cn/didi/didi_2.git/info/lfs.locksverify true
Connection to code.org.cn closed by remote host.                                                                                                                                               
Uploading LFS objects:   0% (0/1), 7.9 GB | 3.0 MB/s  
```

The files info as follows:
```
(base) ➜  didi_2 git:(main) ✗ du -sh *
1.0G	file.json
```

### Gitea Version

1.21.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

Gitea deployed through k8s cluster

### Database

None

**Comment by Zokormazo** [comment]
Is this with minio for lfs backend? I had the same sympthon in https://github.com/go-gitea/gitea/issues/28494

**Comment by alvin6233** [comment]
> Is this with minio for lfs backend? I had the same sympthon in #28494

Yes, I use minio for lfs backend. When I switch to local storage I don't have the problem, with minio I have this problem

**Comment by Zettat123** [comment]
Do you often encounter this bug? I only reproduced it once when the network connection was unstable.

**Comment by yp05327** [comment]
It looks like a network issue. Maybe we can add an option to enable minio http trace.
![image](https://github.com/go-gitea/gitea/assets/18380374/c109df36-dc61-4752-a9c0-3bb68569b38c)

**Comment by stevapple** [comment]
This should be properly solved by buffering (and implementing `ReadAt`), which can allow the MinIO client to re-read some part of the LFS object when recovering from a network problem.

We can choose to utilize cache, memory, disk storage, or a combination of them to balance performance and cost overhead.
